src/Jsend/ServiceStatusResponse.php

```diff
@@ -100,6 +100,72 @@ public function getResponseBody(): array
         ];
     }
 
+    public static function checkServices(
+        array $servicesClients,
+        &$status,
+        ?array $whoAsks = []
+    ): array {
+        $servicesStatus = [];
+        foreach ($servicesClients as $serviceName => $serviceClient) {
+            if (!empty($servicesStatus[$serviceName]) ||
+                is_null($serviceClient) ||
+                in_array($serviceName, $whoAsks)) {
+                $servicesStatus[$serviceName] = StatusEnum::SUCCESS;
+                continue;
+            }
+            $serviceStatusResult = ServiceStatusResponse::getServiceStatus($serviceClient, 'status', $whoAsks);
+            $data = $serviceStatusResult->getResponseBody();
+            if ($data['status'] !== StatusEnum::SUCCESS) {
+                $status = StatusEnum::ERROR;
+            }
+            $servicesStatus[$serviceName] = $data['status'];
+            foreach ($data['data']['services'] as $key => $status) {
+                $servicesStatus[$key] = $status;
+            }
+        }
+        return $servicesStatus;
+    }
+
+    /**
+     * @param ClientInterface $client
+     * @param string $url
+     * @param string|null $serviceName
+     * @return ServiceStatusResponse
+     * @throws \Exception
+     */
+    public static function getServiceStatus(
+        ClientInterface $client,
+        string $url = 'status',
```

Comment:
do we expect any other name for status endpoint than 'status'? Let's make it rather a constant